Sources/Rownd/Models/AppleSignUpCoordinator.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  AppleSignUpCoordinator.swift
+//  appleSignIn
+//
+//  Created by Michael Murray on 7/17/22.
+//
+
+import SwiftUI
+import AuthenticationServices
+import UIKit
+import AnyCodable
+
+fileprivate let appleSignInDataKey = "userAppleSignInData"
+
+struct AppleSignInData: Codable {
+    var email: String
+    var firstName: String?
+    var lastName: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case firstName = "first_name"
+        case lastName = "last_name"
+        case email = "email"
+    }
+}
+
+class AppleSignUpCoordinator: NSObject, ASAuthorizationControllerDelegate, ASAuthorizationControllerPresentationContextProviding {
+    var parent: Rownd?
+    
+    init(_ parent: Rownd) {
+        self.parent = parent
+        super.init()
+    }
+    
+    @objc func didTapButton() {
+        //Create an object of the ASAuthorizationAppleIDProvider
+        let appleIDProvider = ASAuthorizationAppleIDProvider()
+        //Create a request
+        let request         = appleIDProvider.createRequest()
+        //Define the scope of the request
+        request.requestedScopes = [.fullName, .email]
+        //Make the request
+        let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+        
+        //Assigning the delegates
+        authorizationController.presentationContextProvider = self
+        authorizationController.delegate = self
+        authorizationController.performRequests()
+    }
+    
+    func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        let vc = UIApplication.shared.windows.last?.rootViewController
+        return (vc?.view.window!)!
+    }
+    
+    //If authorization is successful then this method will get triggered
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+        switch authorization.credential {
+        case let appleIDCredential as ASAuthorizationAppleIDCredential:
+            
+            // Create an account in your system.
+            //let userIdentifier = appleIDCredential.user
+            let fullName = appleIDCredential.fullName
+            let email = appleIDCredential.email
+            let identityToken = appleIDCredential.identityToken
+            
+            if let email = email {
+                //Store email and fullName in AppleSignInData struct if available
+                let userAppleSignInData = AppleSignInData(email: email, firstName: fullName?.givenName, lastName: fullName?.familyName)
+                let encoder = JSONEncoder()
+                if let encoded = try? encoder.encode(userAppleSignInData) {
+                    let defaults = UserDefaults.standard
```

Comment:
You can just call `Storage.store?.set(encoded, forKey: appleSignInDataKey)` to use the SDK's common storage manager (see Models/Storage.swift)